crytic_compile/platform/hardhat.py

```diff
@@ -132,7 +132,7 @@ def hardhat_like_parsing(
                         )
 
                     source_unit = compilation_unit.create_source_unit(path)
-                    source_unit.ast = info["ast"]
+                    source_unit.ast = info.get("legacyAST", "ast")
```

Comment:
```suggestion
                    source_unit.ast = info.get("legacyAST", info["ast"])
```
Sorry realized my suggestion was incorrect